libs/types/src/cdf/ballot-definition/convert.ts

```diff
@@ -0,0 +1,412 @@
+import {
+  assert,
+  assertDefined,
+  find,
+  naturals,
+  take,
+  throwIllegalValue,
+  unique,
+} from '@votingworks/basics';
+import * as Cdf from '.';
+import * as Vxf from '../../election';
+
+export function convertVxfElectionToCdfBallotDefinition(
+  vxfElection: Vxf.Election
+): Cdf.BallotDefinition {
+  function text(content: string): Cdf.InternationalizedText {
+    return {
+      '@type': 'BallotDefinition.InternationalizedText',
+      Text: [
+        {
+          '@type': 'BallotDefinition.LanguageString',
+          Language: 'en',
+          Content: content,
+        },
+      ],
+    };
+  }
+
+  function dateString(vxDateTimeString: string) {
+    const isoString = new Date(vxDateTimeString).toISOString();
+    return isoString.split('T')[0];
+  }
+
+  function dateTimeString(date: Date) {
+    const isoString = date.toISOString();
+    // Need to remove fractional seconds to satisfy CDF schema
+    return `${isoString.split('.')[0]}Z`;
+  }
+
+  const stateId = vxfElection.state.toLowerCase().replaceAll(' ', '-');
+
+  return {
+    '@type': 'BallotDefinition.BallotDefinition',
+
+    Election: [
+      {
+        '@type': 'BallotDefinition.Election',
+        ElectionScopeId: stateId,
+        StartDate: dateString(vxfElection.date),
+        EndDate: dateString(vxfElection.date),
+        Type: Cdf.ElectionType.General,
```

Comment:
Could this also be `Primary`?

Reply:
Ooh good call. How can I tell from VXF? Maybe looking for contests and/or ballot styles with `partyId`?

Reply:
Ugh, I've been unsure about this. And we've employed two ways looking in the VXF, looking at contests with `partyId` vs. looking at ballot styles with `partyId`. But, because there is the possibility of non-partisan ballots during a primary or an open primary (not supported yet), I think looking at contests is the best bet.

Reply:
Ok, that's helpful. Since we're just using this logic for converting existing fixtures, I'm not too worried about figuring out a long-term solution. In fact, I think this problem will go away once we've moved to CDF.

f37a21beb